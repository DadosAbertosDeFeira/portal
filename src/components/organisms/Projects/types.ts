import type { Project } from "@utils/projects";

export type ProjectsProps = {
  projects: Project[];
};
