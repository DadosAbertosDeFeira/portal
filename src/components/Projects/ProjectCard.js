import React from 'react';
import PropTypes from 'prop-types';
import { FiArrowUpRight } from 'react-icons/fi';

import Tag from '../Tag';
import Link from '../Link';

import styles from './ProjectCard.module.scss';

const ProjectCard = ({ project }) => {
  return (
    <div className={`${styles.card} flex m-md tablet:m-xl rounded`}>
      {/* Logo Wrapper */}
      <div>
        <img
          className={`${styles.icon} rounded-tl rounded-bl`}
          src={project.logo.path}
          alt={project.logo.description}
        />
      </div>

      {/* Content */}
      <div className="flex flex-col pr-xs pb-xs tablet:pr-sm tablet:pt-sm tablet:pb-xs">
        <h3 className="mb-sm tablet:mb-md">{project.name}</h3>
        <p className="text-black text-xs flex-1">{project.description}</p>

        <div className={`flex flex-row ${styles.footer}`}>
          <div>
            {project.type.map((tag) => (
              <Tag key={tag} className="mr-xs">
                {tag}
              </Tag>
            ))}
          </div>

          <Link
            className={`${styles.projectLink} flex items-center`}
            href={project.link}
            target="_blank"
            rel="noreferrer"
          >
            Link
            <FiArrowUpRight className={`ml-xs ${styles.projectLinkIcon}`} />
          </Link>
        </div>
      </div>
    </div>
  );
};

ProjectCard.propTypes = {
  project: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
    link: PropTypes.string,
    logo: PropTypes.shape({
      path: PropTypes.string,
      description: PropTypes.string,
    }),
  }),
};

ProjectCard.defaultProps = {
  project: {},
};

export default ProjectCard;
