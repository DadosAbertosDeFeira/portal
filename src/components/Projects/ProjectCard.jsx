import React from 'react';
import Image from 'next/image';
import PropTypes from 'prop-types';
import { FiArrowUpRight } from 'react-icons/fi';

import Link from 'next/link';
import Tag from '../Tag';

const ProjectCard = ({ project }) => {
  return (
    <div className="max-w-[500px] flex align-center tablet:flex m-md tablet:m-xl rounded shadow-[0px_6px_12px_#0000000d]">
      {/* Logo Wrapper */}
      <div>
        <Image
          className="max-w-[initial] h-auto rounded-tl rounded-bl"
          src={project.logo.path}
          alt={project.logo.description}
        />
      </div>

      {/* Content */}
      <div className="flex flex-col pr-sm pl-sm pb-xs tablet:pr-sm tablet:pt-sm tablet:pb-xs tablet:pl-sm">
        <h3 className="mb-sm tablet:mb-md">{project.name}</h3>
        <p className="text-black text-xs flex-1">{project.description}</p>

        <div className="flex flex-row justify-between">
          <div className="hidden sm:flex">
            {project.type.map((tag) => (
              <Tag key={tag} className="mr-xs">
                {tag}
              </Tag>
            ))}
          </div>

          <Link
            className="text-primary"
            href={project.link}
            target="_blank"
            rel="noreferrer"
          >
            <div className="flex items-center">
              Link
              <FiArrowUpRight className="ml-xs stroke-primary" />
            </div>
          </Link>
        </div>
      </div>
    </div>
  );
};

ProjectCard.propTypes = {
  project: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.arrayOf(PropTypes.string),
    description: PropTypes.string,
    link: PropTypes.string,
    logo: PropTypes.shape({
      // eslint-disable-next-line react/forbid-prop-types
      path: PropTypes.object,
      description: PropTypes.string,
    }),
  }),
};

ProjectCard.defaultProps = {
  project: {},
};

export default ProjectCard;
