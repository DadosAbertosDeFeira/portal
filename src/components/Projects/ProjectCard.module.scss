@import '../../styles/variables.scss';

.card {
  max-width: 500px;
  box-shadow: 0px 6px 12px rgba(0, 0, 0, 0.05);
}

.icon {
  max-width: initial;
  height: auto;
}

.footer {
  justify-content: space-between;
}

.projectLink {
  color: $color-primary;
}

.projectLinkIcon {
  stroke: $color-primary;
}
