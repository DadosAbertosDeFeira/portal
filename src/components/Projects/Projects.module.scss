.icon {
  height: 100px;
  width: 100px;
}

.contentWrapper {
  max-width: 1240px;
}
