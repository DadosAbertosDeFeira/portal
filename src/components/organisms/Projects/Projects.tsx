import IconProject from "@assets/icons/icon-projects.svg";
import { Image } from "atoms/Image";
import { Link } from "atoms/Link";
import { TitleSection } from "molecules/TitleSection";
import React from "react";
import { BiLinkExternal } from "react-icons/bi";

import type { ProjectsProps } from "./types";

export const Projects = ({ projects }: ProjectsProps) => {
  return (
    <TitleSection
      className="p-2 lg:px-20"
      title="Projetos"
      src={IconProject}
      alt=""
    >
      <ul className="grid auto-rows-auto grid-cols-[repeat(auto-fill,_minmax(250px,_1fr))] items-center justify-items-center gap-8 md:grid-cols-[repeat(auto-fill,_minmax(450px,_1fr))]">
        {projects.map(({ name, description, link, type, logo }) => (
          <li className="h-full w-full" key={name}>
            <Link className="h-full w-full" href={link}>
              <section className="relative flex h-full w-full flex-row gap-2 overflow-hidden rounded shadow-lg">
                <Image
                  className="hidden h-[140px] max-h-[140px] min-h-[140px] w-[140px] min-w-[140px] max-w-[140px] object-cover md:block"
                  src={logo.path}
                  alt={logo.description}
                />
                <div className="mt-2 flex h-full flex-col gap-2 p-2">
                  <h3 className="pr-12 text-base">{name}</h3>
                  <p className="overflow-hidden text-xs leading-tight">
                    {description}
                  </p>
                  <ul className="mb-2 flex grow flex-row flex-wrap items-end gap-1 text-xs">
                    {type.map((item) => (
                      <li
                        className="shrink rounded bg-gray-50 p-1 capitalize text-gray-400"
                        key={item}
                      >
                        {item}
                      </li>
                    ))}
                  </ul>
                  <p className="absolute right-1 top-1 flex flex-row flex-nowrap items-center gap-1 text-sm text-blue-400">
                    Link
                    <BiLinkExternal fill="#3793df" />
                  </p>
                </div>
              </section>
            </Link>
          </li>
        ))}
      </ul>
    </TitleSection>
  );
};
