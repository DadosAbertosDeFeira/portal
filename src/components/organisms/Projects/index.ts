export * from "./Projects";
export * from "./types";
