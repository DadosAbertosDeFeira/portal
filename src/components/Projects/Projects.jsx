import React from 'react';
import Image from 'next/image';

import projects from '../../utils/projects';
import IconProject from '../../../public/assets/icons/icon-projects.svg';

import ProjectCard from './ProjectCard';
import styles from './Projects.module.scss';

const Projects = () => {
  return (
    <div>
      {/* Header */}
      <div className="flex flex-col items-center">
        <Image
          className={`${styles.icon} mb-md`}
          src={IconProject}
          alt="Logo da sessÃ£o de projetos"
        />
        <h2>Projetos</h2>
      </div>

      {/* Content */}
      <div
        className={`${styles.contentWrapper} flex flex-col tablet:flex-row tablet:flex-wrap tablet:justify-center tablet:content-evenly mt-md`}
      >
        {projects.map((project) => (
          <ProjectCard key={project.name} project={project} />
        ))}
      </div>
    </div>
  );
};

export default Projects;
