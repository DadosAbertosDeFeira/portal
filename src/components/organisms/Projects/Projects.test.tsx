import { render } from "@testing-library/react";
import type { Project } from "@utils/projects";
import type { ImageProps } from "atoms/Image";

import { Projects } from "./Projects";
import type { ProjectsProps } from "./types";

jest.mock("next/image", () => ({ src, ...props }: ImageProps) => (
  <img src={src as string} {...props} />
));

describe("Projects", () => {
  const projectsMock: Project[] = [
    {
      description:
        "Tem a missão de libertar dados do município de Feira de Santana. Responsável pela raspagem e o armazenamento.",
      img: {
        src: "https://test.com/image.png",
        alt: "Logo do Projeto Maria Quitéria",
      },
      link: "https://github.com/DadosAbertosDeFeira/maria-quiteria",
      name: "Maria Quitéria",
      type: ["raspagem de dados", "busca de dados"],
    },
    {
      description:
        "Tem a missão de libertar dados do município de Feira de Santana. Responsável pela raspagem e o armazenamento.",
      img: {
        src: null,
        alt: "Logo do Projeto Maria Quitéria",
      },
      link: "https://github.com/DadosAbertosDeFeira/maria-quiteria",
      name: "Maria Quitéria",
      type: ["raspagem de dados", "busca de dados"],
    },
  ];

  const makeSut = ({ projects = projectsMock }: Partial<ProjectsProps>) => {
    return render(<Projects projects={projects} />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });
});
