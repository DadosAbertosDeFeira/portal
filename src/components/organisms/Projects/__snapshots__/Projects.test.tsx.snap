// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders correctly 1`] = `[Function]`;
